nodewatcher/modules/monitor/http/survey/processors.py

```diff
@@ -0,0 +1,105 @@
+from nodewatcher.modules.monitor.sources.http import processors as http_processors
+from nodewatcher.modules.monitor.datastream import fields as ds_fields, models as ds_models
+from nodewatcher.modules.monitor.datastream.pool import pool as ds_pool
+
+
+class SurveyInfoStreams(ds_models.RegistryRootStreams):
+    neighbor_graph = ds_fields.GraphField(tags={
+        'title': _("Neighborhood topology"),
+        'description': _("Neighborhood topology."),
+    })
+
+    def get_module_name(self):
+        return 'monitor.http.survey'
+
+    def get_stream_highest_granularity(self):
+        # The nodewatcher-agent performs a survey once every ~240 monitoring intervals according to
+        # https://github.com/wlanslovenija/nodewatcher-agent/blob/master/modules/wireless.c#L362
+        # and a monitoring run is performed every 30 seconds according to
+        # https://github.com/wlanslovenija/nodewatcher-agent/blob/73f2b25db2c34ae7d70904bac31379a2243bade0/modules/wireless.c#L437.
+        # So a survey is performed once every two hours, meaning that we use hourly granularity.
+        return datastream.Granularity.Hours
+
+
+class SurveyInfoStreamsData(object):
+    def __init__(self, node, neighbor_graph):
+        self.node = node
+        self.neighbor_graph = neighbor_graph
+
+
+ds_pool.register(SurveyInfoStreamsData, SurveyInfoStreams)
+
+
+class SurveyInfo(monitor_processors.NodeProcessor):
+    """
+    Accesses the data collected during a WiFi survey of neighboring wireless access points / routers. Stores channel,
+    BSSID, SSID and signal strength for each neighbor into the datastream. A star shaped graph is constructed. The
+    source vertex contains an array of all BSSIDs of that node. Source vertex is stored according to its UUID,
+    wherease all neighbors are stored according to their BSSIDs. Will only run if HTTP monitor module has previously
+    fetched data.
+    """
+
+    @monitor_processors.depends_on_context("http", http_processors.HTTPTelemetryContext)
+    def process(self, context, node):
+        """
+        Called for every processed node.
+
+        :param context: Current context
+        :param node: Node that is being processed
+        :return: A (possibly) modified context
+        """
+
+        version = context.http.get_module_version('core.wireless')
+        if version == 0:
+            # Unsupported version or data fetch failed
+            return context
+
+        edges, vertices = [], []
+        try:
+            source_vertex_bssids = [interface.bssid for interface in context.http.core.wireless.interfaces.values()]
+            source_vertex = {
+                'i': str(node.uuid),
+                'bssids': source_vertex_bssids,
+            }
+            vertices.append(source_vertex)
+            for radio in context.http.core.wireless.radios.values():
+                for neighbor in radio.survey:
+                    vertices.append({'i': neighbor['bssid']})
```

Comment:
My understanding is that you need two antennas to broadcast on two different frequencies. And each antenna has a BSSID associated with it. Otherwise your computer couldn't distinguish between connecting to the 2.4 or 5GHz band? For instance, look at the interfaces at https://github.com/CdavM/swoon/blob/master/test_data/10.20.32.40.feed 
wlan0 (5GHz) has a bssid of 04:18:D6:21:60:44 and wlan1 (2.4GHz) has a bssid of 04:18:D6:22:60:44